src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/TseAdminService.java

```diff
@@ -0,0 +1,329 @@
+import uk.gov.hmcts.et.common.model.ccd.items.RespondentSumTypeItem;
+import uk.gov.hmcts.et.common.model.ccd.items.TseAdminRecordDecisionTypeItem;
+import uk.gov.hmcts.et.common.model.ccd.types.TseAdminRecordDecisionType;
+import uk.gov.hmcts.et.common.model.ccd.types.UploadedDocumentType;
+import uk.gov.hmcts.ethos.replacement.docmosis.utils.IntWrapper;
+import uk.gov.hmcts.ethos.replacement.docmosis.utils.TSEAdminEmailRecipientsData;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.YES;
+import static uk.gov.hmcts.ethos.replacement.docmosis.helpers.TseHelper.formatAdminReply;
+import static uk.gov.hmcts.ethos.replacement.docmosis.helpers.TseHelper.formatRespondentReplyForDecision;
+import static uk.gov.hmcts.ethos.replacement.docmosis.helpers.TseHelper.getSelectedApplicationTypeItem;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+@SuppressWarnings({"squid:S1192", "PMD.AvoidInstantiatingObjectsInLoops"})
+public class TseAdminService {
+
+    @Value("${tse.admin.record-a-decision.claimant.template.id}")
+    private String emailToClaimantTemplateId;
+    @Value("${tse.admin.record-a-decision.respondent.template.id}")
+    private String emailToRespondentTemplateId;
+
+    private final EmailService emailService;
+    private final DocumentManagementService documentManagementService;
+
+    private static final String RESPONSE_APP_DETAILS = "| | |\r\n"
+            + "|--|--|\r\n"
+            + "|%s application | %s|\r\n"
+            + "|Application date | %s|\r\n"
+            + "|%s | %s|\r\n"
+            + "|Supporting material | %s|\r\n"
+            + "\r\n";
+
+    private static final String CLOSE_APP_DETAILS = "| | |\r\n"
+        + "|--|--|\r\n"
+        + "|Applicant | %s|\r\n"
+        + "|Type of application | %s|\r\n"
+        + "|Application date | %s|\r\n"
+        + "|What do you want to tell or ask the tribunal? | %s|\r\n"
+        + "|Supporting material | %s|\r\n"
+        + "|Do you want to copy this correspondence to the other party to satisfy the Rules of Procedure? | %s|\r\n"
+        + "\r\n";
+
+    private static final String CLOSE_APP_DECISION_DETAILS = "|Decision | |\r\n"
+        + "|--|--|\r\n"
+        + "|Notification | %s|\r\n"
+        + "|Decision | %s|\r\n"
+        + "|Date | %s|\r\n"
+        + "|Sent by | %s|\r\n"
+        + "|Type of decision | %s|\r\n"
+        + "|Additional information | %s|\r\n"
+        + "|Description | %s|\r\n"
+        + "|Document | %s|\r\n"
+        + "|Decision made by | %s|\r\n"
+        + "|Name | %s|\r\n"
+        + "|Sent to | %s|\r\n"
+        + "\r\n";
+
+    private static final String BOTH = "Both parties";
+    private static final String CLAIMANT_ONLY = "Claimant only";
```

Comment:
i have seen these several times now. I would recommend these be pulled out into a constants class like in et-common.

Reply:
or even put it in et common